irods_consortium_continuous_integration_test_hook.py

```diff
@@ -19,17 +19,17 @@ def kdc_database_master_key():
 def unprivileged_principal_password():
     return 'krbtest'
 
+def add_alias_to_etc_hosts():
+    alias = '.'.join(['icat', 'example', 'org'])
 
-def add_shortname_to_etc_hosts():
-    fullname = socket.gethostname()
-    shortname = fullname.partition('.')[0]
     with tempfile.NamedTemporaryFile() as hosts_copy:
         with open('/etc/hosts', 'r') as hosts_file:
             for l in hosts_file:
-                if fullname in l:
-                    hosts_copy.write(l.strip() + ' ' + shortname + '\n')
+                # add the alias to the localhost line so that the server self-identifies as the KDC host
+                if '127.0.0.1' in l and alias not in l:
+                    hosts_copy.write('{} {}\n'.format(l.strip(), alias).encode())
                 else:
-                    hosts_copy.write(l)
+                    hosts_copy.write(l.encode())
```

Comment:
Just wondering... Should `encode()` take the encoding as a string or does it default to `utf-8` or some other value?

Reply:
https://docs.python.org/2/library/stdtypes.html?highlight=encode#str.encode

In python2, `encoding` is an optional parameter which defaults to "the current default string encoding." In python3, `encoding` defaults to "utf-8" explicitly, so I guess we will need to indicate this for the 4-2-stable version.

Reply:
Done